internal/registryutil/auth.go

```diff
@@ -18,12 +18,16 @@ package registryutil
 import (
 	"context"
 
-	"oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/registry/remote"
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
 // WithScopeHint adds a hinted scope to the context.
```

Comment:
Do we need to mention that this only applies to repositories?

Reply:
It's optional since this function is already internal. Also, if it's misused on a non-repo target like OCI layout or memory, there is no side effects and the performance cost is ignorable.